pycloudlib/ibm_softlayer/instance.py

```diff
@@ -0,0 +1,374 @@
+
+        """
+        logger.debug("Creating raw instance")
+        constant_args = {
+            "private": False,
+            "dedicated": False,  # default
+            "hourly": True,  # default
+            "local_disk": False,  # default
+        }
+        instance_specific_args = {
+            "domain": domain_name,
+            "hostname": hostname,
+            "ssh_keys": ssh_key_ids,
+            "image_id": target_image_global_identifier,
+            "flavor": flavor,
+            "datacenter": datacenter,
+            "public_security_groups": public_security_group_ids,
+            "private_security_groups": private_security_group_ids,
+        }
+        # check if instance configuration is valid
+        try:
+            logger.info(
+                "Verifying configuration for instance before creating it."
+            )
+            vs_manager.verify_create_instance(
+                **constant_args,
+                **instance_specific_args,
+                **kwargs,
+            )
+        except SoftLayer.SoftLayerAPIError as e:
+            logger.error("configuration for instance is invalid: %s", e)
+            raise IBMSoftlayerException(
+                f"Failed to verify instance configuration: {e}"
+            ) from e
+        except Exception as e:
+            logger.error(
+                "Unexpected error while verifying instance configuration: %s",
+                e,
+            )
+            raise IBMSoftlayerException(
+                f"Unexpected error while verifying instance configuration: {e}"
+            ) from e
+        else:
+            logger.info(
+                "Configuration for instance is valid. Creating instance now."
+            )
+            raw_instance = vs_manager.create_instance(
+                **constant_args,
+                **instance_specific_args,
+                **kwargs,
+            )
+            logger.info("Created instance %s", raw_instance["hostname"])
+            full_raw_instance_info = vs_manager.get_instance(
+                raw_instance["id"]
+            )
+            logger.debug("New instance details: %s", full_raw_instance_info)
+            return full_raw_instance_info
+
+    def get_boot_id(self):
+        """Get the instance boot_id.
+
+        Returns:
+            string with the boot UUID
+        """
+        result = self.execute(
+            "cat /proc/sys/kernel/random/boot_id", no_log=True
+        )
+        if result.failed:
+            raise OSError(
+                f"Failed to get boot_id. Return code: {result.return_code}, "
+                f"stdout: {result.stdout}, stderr: {result.stderr}"
+            )
+        return result
+
+    def console_log(self):
+        """Return the instance console log.
+
+        Raises NotImplementedError if the cloud does not support fetching the
+        console log for this instance.
+        """
+        # IBM Classic does not support console log
```

Comment:
Nit: This comment is already documented in the exception. It feels redundant.

```suggestion
```

Reply:
hahaha agreed